linkerd/app/src/trace_collector/otel_collector.rs

```diff
@@ -38,37 +31,33 @@ where
     let (span_sink, spans_rx) = mpsc::channel(crate::trace_collector::SPAN_BUFFER_CAPACITY);
     let spans_rx = ReceiverStream::new(spans_rx);
 
-    let mut resources = ResourceAttributesWithSchema::default();
-
-    resources
-        .attributes
-        .0
-        .push((std::process::id() as i64).with_key(semconv::attribute::PROCESS_PID));
-
-    resources.attributes.0.push(
-        SystemTime::now()
-            .duration_since(UNIX_EPOCH)
-            .map(|d| d.as_secs() as i64)
-            .unwrap_or_else(|e| -(e.duration().as_secs() as i64))
-            .with_key("process.start_timestamp"),
-    );
-    resources.attributes.0.push(
-        attributes
-            .hostname
-            .unwrap_or_default()
-            .with_key(semconv::attribute::HOST_NAME),
-    );
-
-    resources.attributes.0.extend(
-        attributes
-            .extra
-            .into_iter()
-            .map(|(key, value)| value.with_key(&key)),
-    );
+    let resource = sdk::Resource::builder()
+        .with_attribute(KeyValue::new(
+            semconv::attribute::PROCESS_PID,
+            std::process::id() as i64,
```

Comment:
This may be a good time to future-proof potential pid space growth?

```suggestion
            i64::from(std::process::id()),
```